tests/v2/test_enr_utils.nim

```diff
@@ -0,0 +1,65 @@
+{.used.}
+
+import
+  testutils/unittests,
+  std/options,
+  stew/byteutils,
+  chronos,
+  ../../waku/v2/utils/wakuenr,
+  ../test_helpers
+
+procSuite "ENR utils":
+
+  asyncTest "Parse multiaddr field":
+    let
+      reasonable = "0x000A047F0000010601BADD03".hexToSeqByte()
+      tooLong = "0x000B047F0000010601BADD03".hexToSeqByte()
+      tooShort = "0x000A047F0000010601BADD0301".hexToSeqByte()
+      gibberish = "0x3270ac4e5011123c".hexToSeqByte()
+      empty = newSeq[byte]()
+    
+    ## Note: we expect to fail optimistically, i.e. extract
+    ## any addresses we can and ignore other errors.
+    ## Worst case scenario is we return an empty `multiaddrs` seq.    
+    check:
+      # Expected case
+      reasonable.toMultiAddresses().contains(MultiAddress.init("/ip4/127.0.0.1/tcp/442/ws")[])
+      # Buffer exceeded
+      tooLong.toMultiAddresses().len() == 0
+      # Buffer remainder
+      tooShort.toMultiAddresses().contains(MultiAddress.init("/ip4/127.0.0.1/tcp/442/ws")[])
+      # Gibberish
+      gibberish.toMultiAddresses().len() == 0
+      # Empty
+      empty.toMultiAddresses().len() == 0
+
+  asyncTest "Init ENR for Waku Usage":
+    let
+      enrIp = ValidIpAddress.init("127.0.0.1")
+      enrTcpPort, enrUdpPort = Port(60000)
+      enrKey = PrivateKey.random(Secp256k1, rng[])[]
+      wakuFlags = initWakuFlags(false, true, false, true)
+      multiaddrs = @[MultiAddress.init("/ip4/127.0.0.1/tcp/442/ws")[], MultiAddress.init("/ip4/127.0.0.1/tcp/443/wss/p2p/16Uiu2HAm4v86W3bmT1BiH6oSPzcsSr31iDQpSN5Qa882BCjjwgrD")[]]
```

Comment:
[31/WAKU2-ENR](https://rfc.vac.dev/spec/31/) defines that no peer id should be encoded in the `mutliaddrs` field. Hence I would recommend to not have a peer id in the happy path test.

If you want to stripe the peer id and test it then I'd do a separate test for it.

Reply:
Makes sense. Done in fffcdb2.